Adds hue_sensors

robmarkcole opened the pull request:
## Description:
Component for reading Hue motion sensors and 4 button remotes.

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io/pull/3380) with documentation (if applicable):** home-assistant/home-assistant.github.io#3380

## Example entry for `configuration.yaml` (if applicable):
```yaml
hue:

sensor:
  - platform: hue
```

## Checklist:

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) NEEDS UPDATING

If the code communicates with devices, web services, or third-party tools:
  - [x] Local tests with `tox` run successfully. **Your PR cannot be merged unless tests pass**

robmarkcole committed:
Adds hue_sensors

Component for reading Hue motion sensors and 4 button remotes

robmarkcole committed:
remove re

robmarkcole committed:
Fix lints

robmarkcole committed:
Fix more lints

robmarkcole committed:
Use iter

robmarkcole commented:
I am using the same approach as `/homeassistant/components/light/hue.py` so not sure why there is an error `consider-iterating-dictionary`

pvizeli commented:
You need convert hue into a component/platform solution

robmarkcole commented:
Hi @pvizeli can you elaborate on what conversion to a component/platform solution involves, or point to some docs/example?
Cheers

robmarkcole committed:
Updates state of motion sensors

Updates state of motion sensors to be ‘on’ or ‘off’ as per the
convention for binary sensors

robmarkcole committed:
Updates icons

robmarkcole committed:
Adds Hue tap

robmarkcole committed:
Fix lint

fanaticDavid commented:
Just a small detail, but wouldn't it be better to use the name `hue` instead of `hue_sensors`? The `sensor` component already makes it clear you're dealing with sensors. And it ligns it up with the Hue light component:
```yaml
light:
  - platform: hue
```

balloob commented:
Yep, you will need to split it auth @robmarkcole . Create a hue component that does the auth and stores the bridge in `hass.data[DOMAIN]`. All platforms are required to have the same name and use the bridge object from `hass.data['hue']`

robmarkcole commented:
@balloob I take it that the [hue lights component ](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/light/hue.py) will also need to be modified to use the auth model? Can you suggest a good example for this model? 
I also noticed that the docs [don't describe this model ](https://home-assistant.io/developers/platform_example_light/)unfortunately, unless I am mistaken?
Cheers

@rossdargan

balloob commented:
Yes, that would be okay.

The example you linked is for an integration with just a platform. We do not have an example with auth and 2 platforms.

robmarkcole committed:
Converts hue into a component/platform solution

Spilts out hue hub into its own component which fetches the URL for API
requests. This is likely a work in progress.

robmarkcole committed:
Fix imports

robmarkcole committed:
Respond to Travis

drop .keys() usage and replace self.ICON with self._icon

balloob commented on homeassistant/components/sensor/hue.py:
```diff
@@ -0,0 +1,195 @@
+    return data_dict
+
+
+def parse_sml(response):
```
This is too much protocol specific information. Please extract this to a new 3rd party library

robmarkcole committed:
Imports hue-sensors from pypi

Still working on the tests

robmarkcole committed:
Adds basic test

Need a more comprehensive test - don’t see tests for the hue lights?
Another good example?

robmarkcole commented:
Need to improve test. Was hoping to use hue lights as an example, but can't find? 
Alternatively [dark_sky](https://github.com/robmarkcole/home-assistant/blob/dev/tests/components/sensor/test_darksky.py) looks appropriate.

andrey-git reviewed (commented):
Robin, please add yourself to `CODEOWNERS` or the new files.

I think the json test file belong more in the lib. Home Assistant should mock the lib in the tests if possible.

robmarkcole commented:
Hi @andrey-git do you mean that the json test file should be in the external lib hue-sensors? Cheers

robmarkcole committed:
Updates CODEOWNERS

andrey-git commented on CODEOWNERS:
```diff
@@ -11,6 +11,7 @@ homeassistant/components/automation/* @home-assistant/core
 homeassistant/components/configurator.py @home-assistant/core
 homeassistant/components/group.py @home-assistant/core
 homeassistant/components/history.py @home-assistant/core
+homeassistant/components/hue.py @robmarkcole
```
Sort alphabetically

andrey-git commented:
I don't understand why the JSON file is loaded at all (didn't see it name mentioned anywhere)

In any case as I understand this file represents a mock of `requests`response.
Instead you should mock `hue-sensors`

robmarkcole committed:
Correct order of CODEOWNERS

Sorted alphabetically

robmarkcole commented:
HI @andrey-git I don't quite understand. The HueSensorData object is updated via a request, and it is the response of this which I mock. The response is then parsed by parse_hue_api_response() which is imported from `hue-sensors`. Can you recommend a component which implements the approach you recommend?

Btw my repo is now in a 'detached head' state so I may close this pull and open a fresh one

balloob commented:
You should mock `parse_hue_api_response` with the result that it would produce from the JSON.

The actual raw JSON being converted to what it produces is part of the hue sensors package.

robmarkcole commented:
Having problems with my 'detached head' so will start a new PR
